### PR TITLE
Update names for internal BV operators

### PR DESCRIPTION
**Pull request by ajreynol**
We requiring prefixing internal operators with `@` to avoid name clashes with SMT-LIB user symbols.
 
Also does some cleaning of the ALF signature for these operators.

### CODE REVIEW

**Review by HanielB — COMMENTED** [review]
This will break the Alethe backend in proof-new. So please do not merge into that before we work out how this will work with Alethe and Carcara.

**Comment by vleni** [comment]
In `rw_parser.py`, the list of special chars in symbols must be extended for this to work:
```python
    def symbol(self):
        special_chars = '=' + '_' + '+' + '-' + '<' + '>' + '*' + '.' + '@'
        return pp.Word(pp.alphas + special_chars, pp.alphanums + special_chars)
```

**Comment by ajreynol** [comment]
Ok, can you include this your other PR and I will rebase?

**Comment by vleni** [comment]
> Ok, can you include this your other PR and I will rebase?

what do you mean by include this your other PR

File: proofs/alf/cvc5/theories/BitVectors.smt3 [hunk 1]
```diff
@@ -3,15 +3,22 @@
 (declare-const BitVec (-> Int Type))
 (declare-consts <binary> (BitVec (alf.len alf.self)))
 
+(program get_bitwidth ((T Type) (n Int))
```
Comment (abdoo8080) [inline proofs/alf/cvc5/theories/BitVectors.smt3:6]: Is `(T Type)` needed?

File: proofs/alf/cvc5/theories/BitVectors.smt3 [hunk 2]
```diff
@@ -304,9 +299,21 @@
...
+(declare-const @bbT (->
+  (! Int :var n :implicit)
+  Bool
+  (BitVec n)
+  (BitVec (alf.add 1 n))) :right-assoc-nil @bvempty
+)
+
 ; symbolic constant
-;(declare-const bv (-> Int (! Int :var w) (BitVec w)))
+(declare-const @bv (-> Int (! Int :var w) (BitVec w)))
```
Comment (abdoo8080) [inline proofs/alf/cvc5/theories/BitVectors.smt3:316]: How is this different from an uninterpreted constant?

File: proofs/alf/cvc5/theories/BitVectors.smt3 (hunk 1 above)
Comment (ajreynol) [reply proofs/alf/cvc5/theories/BitVectors.smt3:6]: No, good catch.

File: proofs/alf/cvc5/theories/BitVectors.smt3 (hunk 2 above)
Comment (ajreynol) [reply proofs/alf/cvc5/theories/BitVectors.smt3:316]: For now, it is essentially an uninterpreted function.  However, we will add an evaluator for it that transforms @bv to a binary constant.

**Review by abdoo8080 — APPROVED** [review]
LGTM!

**Pull request merged**